scripts/vmss/common-functions.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+function get_vmss() {
+    #MS az graph query to find and return a list of all VMSS tagged to be included in the auto-shutdown process.
+    log "----------------------------------------------"
+    log "Running az graph query..."
+
+    if [ -z $1 ]; then
+        env_selector=""
+    elif [ $1 == "untagged" ]; then
+        env_selector="| where isnull(tags.environment) and isnull(tags.Environment)"
+    else
+        env_selector="| where tags.environment contains '$1' or tags.Environment contains '$1'"
+    fi
+
+    if [ -z $2 ]; then
+        area_selector=""
+    else
+        area_selector="| where tolower(tags.businessArea) == tolower('$2')"
+    fi
+
+   az graph query -q "
+    resources
+    | where type =~ 'Microsoft.Compute/virtualMachineScaleSets'
+    | where subscriptionId in ('7a4e3bd5-ae3a-4d0c-b441-2188fee3ff1c', '1c4f0704-a29e-403d-b719-b90c34ef14c9', 'bf308a5c-0624-4334-8ff8-8dca9fd43783')
+    | where tags.autoShutdown == 'test'
+    | where not (name matches regex '(^aks-|-aks-|-aks$)')
+    | where not (resourceGroup matches regex '(^aks-|-aks-|-aks$)')
```

Comment:
Line 27 & 28 are doing the same thing, are they both needed?

Reply:
line 27 is to exclude resources with AKS in the name and line 28 is the equivalent but for RG's, so both are needed.

Reply:
Its to ensure that VMSS's that are part of the AKS clusters are not included in the auto shutdown, currently we'rve restricted the script to DCD-CNP-QA, DCD-CNP-DEV & DCD-CFT-Sandbox for now. So it covers us when we open it up to further subscriptions, to capture resources groups or resources with that name aks.